internal/services/securitycenter/security_center_server_vulnerability_assessment_virtual_machine_resource.go

```diff
@@ -0,0 +1,184 @@
+	d.SetId(id.ID())
+
+	timeout, _ := ctx.Deadline()
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending:      []string{"Pending"},
+		Target:       []string{"Succeeded"},
+		Refresh:      serverVulnerabilityAssessmentVirtualMachineStateRefreshFunc(ctx, client, id.ResourceGroup, id.VirtualMachineName),
+		PollInterval: 10 * time.Second,
+		Timeout:      time.Until(timeout),
+	}
+
+	if _, err := stateConf.WaitForStateContext(ctx); err != nil {
+		return fmt.Errorf("waiting for the completion of the creating/updating of %s: %+v", id, err)
+	}
+
+	return resourceServerVulnerabilityAssessmentVirtualMachineRead(d, meta)
+}
+
+func resourceServerVulnerabilityAssessmentVirtualMachineRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VulnerabilityAssessmentVmID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	var virtualMachineId *computeParse.VirtualMachineId
+	if id.ID() != "" {
+		newVirtualMachineId := computeParse.NewVirtualMachineID(id.SubscriptionId, id.ResourceGroup, id.VirtualMachineName)
+		virtualMachineId = &newVirtualMachineId
+	} else {
+		newVirtualMachineId, err := computeParse.VirtualMachineID(d.Get("virtual_machine_id").(string))
+		if err != nil {
+			return err
+		}
+		virtualMachineId = newVirtualMachineId
+	}
+
+	resp, err := client.Get(ctx, virtualMachineId.ResourceGroup, computeProvider, vmType, virtualMachineId.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("Server Vulnerability Assessment was not found for %s: %+v", virtualMachineId, err)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Server Vulnerability Assessment status for %s: %+v", virtualMachineId, err)
+	}
+
+	d.SetId(id.ID())
+	d.Set("virtual_machine_id", virtualMachineId.ID())
+
+	return nil
+}
+
+func resourceServerVulnerabilityAssessmentVirtualMachineDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VulnerabilityAssessmentVmID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Cannot delete if still in provisioning state. Wait for it to complete.
+	timeout, _ := ctx.Deadline()
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending:      []string{"Pending"},
+		Target:       []string{"Succeeded"},
+		Refresh:      serverVulnerabilityAssessmentVirtualMachineStateRefreshFunc(ctx, client, id.ResourceGroup, id.VirtualMachineName),
+		PollInterval: 10 * time.Second,
+		Timeout:      time.Until(timeout),
+	}
+
+	if _, err := stateConf.WaitForStateContext(ctx); err != nil {
+		return fmt.Errorf("waiting for the completion of the creating/updating of %s: %+v", id, err)
+	}
```

Comment:
(as above) this would signify an issue elsewhere that we need to track down, this needs to be removed

```suggestion
```